client/src/main/java/io/pravega/client/stream/impl/Controller.java

```diff
@@ -87,16 +88,40 @@
 
     /**
      * API to merge or split stream segments.
+     *
+     * @param stream Stream object.
+     * @param sealedSegments List of segments to be sealed.
+     * @param newKeyRanges Key ranges after scaling the stream.
+     * @return A future which will throw if the operation fails, otherwise returning a boolean to
+     *         indicate that the scaling was started or not.
+     */
+    CompletableFuture<Boolean> startScale(final Stream stream, final List<Integer> sealedSegments,
+                                           final Map<Double, Double> newKeyRanges);
+
+    /**
+     * API to merge or split stream segments.
```

Comment:
Here is what is being done:
RPC: startScale, checkScaleStatus
Client Apis: 
Future startScale()
Future checkScaleStatus()
Future scaleStream()

scaleStream does following: 
1. startStream
2. loop (condition: scale-complete || timeout) --> delayedFuture.checkStatus